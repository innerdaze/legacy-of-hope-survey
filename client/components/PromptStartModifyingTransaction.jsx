import React, { Component } from 'react'
import Layer from 'grommet/components/Layer'
import Header from 'grommet/components/Header'
import Heading from 'grommet/components/Heading'
import Title from 'grommet/components/Title'
import Paragraph from 'grommet/components/Paragraph'
import Footer from 'grommet/components/Footer'
import Anchor from 'grommet/components/Anchor'
import CloseIcon from 'grommet/components/icons/base/Close'
import NextLinkIcon from 'grommet/components/icons/base/LinkNext'

export default ({
  onSubmit = Function.prototype,
  onCancel = Function.prototype
}) => (
  <Layer>
    <Header>
      <Title>
        Not so fast!
      </Title>
    </Header>
    <Heading tag='h2'>
      There is already an entry for this product.
    </Heading>
    <Paragraph>
      Would you like to change the quantity?
    </Paragraph>
    <Footer
      pad={{ vertical: 'medium' }}
      justify='between'
      >
      <Anchor
        label='Cancel'
        icon={<CloseIcon/>}
        onClick={onCancel}
        />
      <Anchor
        primary
        label='Continue'
        icon={<NextLinkIcon/>}
        onClick={onSubmit}
        />
    </Footer>
  </Layer>
)
