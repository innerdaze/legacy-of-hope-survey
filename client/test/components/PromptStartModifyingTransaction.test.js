import React from 'react'
import PromptStartModifyingTransaction from '../../components/PromptStartModifyingTransaction'

let wrapper, onSubmit, onCancel

describe('Testing on PromptStartModifyingTransaction...', () => {
  beforeEach(() => {
    onSubmit = jest.fn()
    onCancel = jest.fn()
    wrapper = shallow(
      <PromptStartModifyingTransaction
        onSubmit={onSubmit}
        onCancel={onCancel}
      />
    )
  })

  describe('PromptStartModifyingTransaction layout', () => {
    test('Expect to have Title, Heading, Paragraph and Footer', () => {
      expect(wrapper.find('Title')).toHaveLength(1)
      expect(wrapper.find('Heading')).toHaveLength(1)
      expect(wrapper.find('Paragraph')).toHaveLength(1)
      expect(wrapper.find('Footer')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })

    test('Expect Footer to have 2 Anchor comps', () => {
      expect(wrapper.find('Anchor')).toHaveLength(2)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect onSubmit method to call if Anchor Continue is clicked', () => {
      wrapper.find('Anchor[label="Continue"]').simulate('click')

      expect(onSubmit).toBeCalled()
    })

    test('Expect onCancel method to call if Anchor Cancel is clicked', () => {
      wrapper.find('Anchor[label="Cancel"]').simulate('click')

      expect(onCancel).toBeCalled()
    })
  })
})
