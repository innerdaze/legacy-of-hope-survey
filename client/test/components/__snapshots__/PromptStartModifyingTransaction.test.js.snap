// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on PromptStartModifyingTransaction... PromptStartModifyingTransaction layout Expect Footer to have 2 Anchor comps 1`] = `
<Layer
  align="center"
>
  <Header
    align="center"
    direction="row"
    pad={
      Object {
        "between": "small",
        "horizontal": "none",
        "vertical": "none",
      }
    }
    responsive={false}
  >
    <Title
      responsive={true}
      truncate={true}
    >
      Not so fast!
    </Title>
  </Header>
  <Heading
    tag="h2"
  >
    There is already an entry for this product.
  </Heading>
  <Paragraph>
    Would you like to change the quantity?
  </Paragraph>
  <Footer
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "vertical": "medium",
      }
    }
    responsive={false}
  >
    <Anchor
      animateIcon={true}
      icon={
        <Close
          responsive={true}
        />
      }
      label="Cancel"
      method="push"
      onClick={[MockFunction]}
      tag="a"
    />
    <Anchor
      animateIcon={true}
      icon={
        <LinkNext
          responsive={true}
        />
      }
      label="Continue"
      method="push"
      onClick={[MockFunction]}
      primary={true}
      tag="a"
    />
  </Footer>
</Layer>
`;

exports[`Testing on PromptStartModifyingTransaction... PromptStartModifyingTransaction layout Expect to have Title, Heading, Paragraph and Footer 1`] = `
<Layer
  align="center"
>
  <Header
    align="center"
    direction="row"
    pad={
      Object {
        "between": "small",
        "horizontal": "none",
        "vertical": "none",
      }
    }
    responsive={false}
  >
    <Title
      responsive={true}
      truncate={true}
    >
      Not so fast!
    </Title>
  </Header>
  <Heading
    tag="h2"
  >
    There is already an entry for this product.
  </Heading>
  <Paragraph>
    Would you like to change the quantity?
  </Paragraph>
  <Footer
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "vertical": "medium",
      }
    }
    responsive={false}
  >
    <Anchor
      animateIcon={true}
      icon={
        <Close
          responsive={true}
        />
      }
      label="Cancel"
      method="push"
      onClick={[MockFunction]}
      tag="a"
    />
    <Anchor
      animateIcon={true}
      icon={
        <LinkNext
          responsive={true}
        />
      }
      label="Continue"
      method="push"
      onClick={[MockFunction]}
      primary={true}
      tag="a"
    />
  </Footer>
</Layer>
`;
